src/mc-fle2-encryption-placeholder.c

```diff
@@ -58,7 +63,7 @@ bool mc_FLE2EncryptionPlaceholder_parse(mc_FLE2EncryptionPlaceholder_t *out,
 
     mc_FLE2EncryptionPlaceholder_init(out);
     if (!bson_validate(in, BSON_VALIDATE_NONE, NULL) || !bson_iter_init(&iter, in)) {
-        CLIENT_ERR("invalid BSON");
+        CLIENT_ERR_PREFIXED("invalid BSON");
```

Comment:
The trailing comma appears to cause MSVC build errors. [Example](https://spruce.mongodb.com/task/libmongocrypt_windows_test_build_and_test_and_upload_patch_0fbf9a0cfb215998c5dbec621c49e554a6344420_65e0c16f3627e0800d1475d0_24_02_29_17_40_00/logs?execution=0):

> mc-fle2-encryption-placeholder.c(66): error C2059: syntax error: ')'

VS Code shows this expands to:
```
_mongocrypt_set_error(status, MONGOCRYPT_STATUS_ERROR_CLIENT, 1, "Error parsing FLE2EncryptionPlaceholder" ": " "invalid BSON",)
```

Suggest using `%s` for use of `CLIENT_ERR_PREFIXED` that only have one argument:
```suggestion
        CLIENT_ERR_PREFIXED("%s", "Invalid BSON");
```

Reply:
So apparently there is a long-standing bug with the MSVC preprocessor which causes this incorrect trailing comma for certain cases of variadic expansion (searching "msvc variadic macro preprocessing bug" yields about 12 years of people complaining about this). One solution is to avoid using macros with only variadic arguments from other macros, which is why I've added the `fmt` leading argument to the CLIENT_ERR macro. This seems to fix the issue for Windows.